core/rpc/client/user/jsonrpc/methods.go

```diff
@@ -36,6 +37,27 @@ func (cl *Client) Broadcast(ctx context.Context, tx *transactions.Transaction, s
 	res := &jsonrpc.BroadcastResponse{}
 	err := cl.call(ctx, string(jsonrpc.MethodBroadcast), cmd, res)
 	if err != nil {
+		var jsonRPCErr *jsonrpc.Error
+		if errors.As(err, &jsonRPCErr) {
+			if jsonRPCErr.Code == jsonrpc.ErrorTxExecFailure {
```

Comment:
Can combine and outdent.

`if errors.As(err, &jsonRPCErr) && jsonRPCErr.Code == jsonrpc.ErrorTxExecFailure {`

Reply:
Could also add an `&& len(jsonRPCErr.Data) > 0`